src/main/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticator.java

```diff
@@ -111,14 +70,18 @@ public HTTPJwtAuthenticator(final Settings settings, final Path configPath) {
         requireIssuer = settings.get("required_issuer");
 
         if (requireAudience != null) {
-            _jwtParser.requireAudience(requireAudience);
+            jwtParserBuilder = jwtParserBuilder.require("aud", requireAudience);
         }
 
         if (requireIssuer != null) {
-            _jwtParser.requireIssuer(requireIssuer);
+            jwtParserBuilder = jwtParserBuilder.require("iss", requireIssuer);
         }
 
-        jwtParser = _jwtParser;
+        if (jwtParserBuilder != null) {
```

Comment:
nit: why do the null check here, but not up above in the require audience and require issuer block. Can the jwtParserBuilder potentially be null?

Reply:
the reason I added this check is that the original credential check logic is kinda different in JWT Authenticator ([source code on main](https://github.com/opensearch-project/security/blob/main/src/main/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticator.java#L144)):
```
...
 private AuthCredentials extractCredentials0(final RestRequest request) {
        if (jwtParser == null) {
            log.error("Missing Signing Key. JWT authentication will not work");
            return null;
        }

        String jwtToken = request.header(jwtHeaderName);
        if (isDefaultAuthHeader && jwtToken != null && BASIC.matcher(jwtToken).matches()) {
            jwtToken = null;
        }
...
```
Since it originally has the logic for checking the `jwtParser`.

Reply:
Ok, moving some of the conversation to here:
> If it is possible to return null from KeyUtil.create...  then I think it makes sense that the check is directly after, otherwise its susceptible to an NPE if a user configures with a required_audience and w/o a signing key.

I have applied the fix.